ui/app/components/clients/history.js

```diff
@@ -89,48 +90,68 @@ export default class History extends Component {
     );
   }
 
-  get latestUpgradeData() {
-    // {id: '1.9.0', previousVersion: null, timestampInstalled: '2021-11-03T10:23:16Z'}
-    // version id is 1.9.0 or earliest upgrade post 1.9.0, timestamp is RFC3339
-    // need to return the last in the array as the data is in ascending order
-    return this.args.model.versionHistory.at(-1) || null;
+  get upgradeVersionHistory() {
+    const versionHistory = this.args.model.versionHistory;
+    if (!versionHistory || versionHistory.length === 0) {
+      return null;
+    }
+
+    // get upgrade data for initial upgrade to 1.9 and/or 1.10
+    let relevantUpgrades = [];
+    const importantUpgrades = ['1.9', '1.10'];
+    importantUpgrades.forEach((version) => {
+      let findUpgrade = versionHistory.find((versionData) => versionData.id.match(version));
+      if (findUpgrade) relevantUpgrades.push(findUpgrade);
+    });
+
+    // if no history for 1.9 or 1.10, get first upgrade available
+    if (Object.entries(relevantUpgrades).length === 0) {
+      relevantUpgrades.push({
+        id: versionHistory[0].id,
+        previousVersion: versionHistory[0].previousVersion,
+        timestampInstalled: versionHistory[0].timestampInstalled,
+      });
+    }
+    // array of upgrade data objects for noteworthy upgrades
+    return relevantUpgrades;
   }
 
-  get allMinorVersionsInRange() {
-    let newArray = [];
-    // ARG TODO testing unsure if the getActivityResponse is the correct one here.
-    let startTime = new Date(this.getActivityResponse.startTime);
-    let endTime = new Date(this.getActivityResponse.endTime);
-    this.args.model.versionHistory.forEach((object) => {
-      let versionDate = new Date(object.timestampInstalled);
-      // compare against the API response to see if date is within the range of returned response.
-      if (isAfter(versionDate, startTime) && isBefore(versionDate, endTime)) {
-        // push the minor version number e.g. 9
-        let upgradeVersionArray = object.id.split('.');
-        newArray.push(upgradeVersionArray[1]);
-      }
+  get upgradeDuringActivity() {
+    const activityStart = new Date(this.getActivityResponse.startTime);
+    const activityEnd = new Date(this.getActivityResponse.endTime);
+    const upgradesWithinData = this.upgradeVersionHistory.filter((upgrade) => {
+      // will timezones be an issue because new Date defaults to current timezone?
+      let upgradeDate = new Date(upgrade.timestampInstalled);
+      return isAfter(upgradeDate, activityStart) && isBefore(upgradeDate, activityEnd);
     });
-    return newArray; // return [9,9,10] shows all the minor numbers that fall within the range of the returned API response.
+    // return all upgrades that happened within date range of queried activity
```

Comment:
We need all the data for upgrades within a date range so we can display the pertinent information in the line chart